### DISCUSSION TITLE
Installation and getting started

### DISCUSSION THREAD

**Discussion by clizlee** [foundation/proton#26] [Q&A]
Ok, I am following the instructions here: https://docs.page/foundation/proton/install and I have composer installed. I'm on a Windows 11 desktop. But when I try to run the next command on the next step `proton init --config` I get an error that I need to be running PHP version ">= 8.1.0". I'm having issues though first switching to that PHP version. I'm running WAMP on my computer already. 

I also tried running composer update or composer install in my project folder but it says to initialize a project please create a composer.json file. So I may be missing a step somewhere am I supposed to clone the Proton repo first? Or am I supposed to install something else? I'm really confused. Then also I may need help updating my PHP version. So not sure if anyone can help. I really just wanted to try to get started using Proton to see how it works. I've been using Panini but wanted to test this out.

**Comment by benthien** [comment]
I am also using wamp. If wampmanager is running there should be an icon in the system tray that is green if wamp is working correctly (red if not working, orange if partly working). If it is not running, you can find wampanager.exe in c:\wamp or c:\wamp64 depending on whether you are using the 32-bit or 64-bit version of wamp. Execute this program to start the manager. A left-click of the icon in the system tray will show the PHP version being used. If this is not the correct version, right-click the icon and select tools>Change PHP CLI version. This will open a dialog box that shows the PHP versions available. If the desired version is there, clicking on it will set this as the version being used. If it is not there, you need to install it. To install the correct version of PHP, right-click the icon and select Help>Go to https://wampserver.aviatechno.net/. This will open a website containing updates to wamp and its add-ons. Find the correct version of PHP and install it. Then make that version active as shown above.

**Comment by benthien** [comment]
I just discovered that the latest upgrade of wamp has changed where you activate a different version of PHP. You left-click the system tray icon and select PHP>Version to pop up the list of available versions,

**Reply by clizlee** [reply]
@benthien thanks. I was still having trouble for some reason so I uninstalled and reinstalled wampserver and I was finally able to get past the first steps.

**Comment by clizlee** [comment]
So now I have this issue after running `proton build`
![image](https://user-images.githubusercontent.com/4943099/226512412-c5ef7a10-199b-49c0-a216-34826918d4b3.png)

I guess that means I don't have yarn installed. But there was no mention of installing yarn in the documentation. Do I install yarn using `npm install --global yarn`? 

My next question is how do you install the foundation sites files with using proton?

**Reply by benthien** [reply]
You do need yarn. Foundation usually doesn't work for all versions of node. I use nvm so I can control the version of node used. Once you have node installed, you use `npm install --global yarn` to install yarn. Starting with beta 5 proton has the command `proton init --template=sites` to set up foundation-sites. Running yarn then adds node modules needed. for foundation-sites. Unfortunately, the test provided doesn't use any foundation commands. I found that the default layout provided didn't work on my foundation sites. I have attached a zip file containing a layout file showing the changes I made.
```

[default.zip](https://github.com/foundation/proton/files/11141163/default.zip)

**Reply by clizlee** [reply]
well I was able to get yarn installed this time I have the correct file setup in my project folder now. but no foundation sites files. Not sure what I am missing. So I just installed the basic template of foundation sites, but that doesn't seem quite right either. I'm currently running node 14.19.1. I've been successfully using Foundation Panini for a while now, so not sure why I'm not able to get this running. I wish the instructions were clearer. It seems like there's a lot missing in the Installation instructions or I am just missing something. Guess I'll just wait til this is further along in development and stick to Panini for now.

**Reply by benthien** [reply]
When you run `proton init --template=sites` in some directory it sets up a directory structure similar to panini's and some setup files. You then run yarn. This will import the node modules. foundation-sites is one of those modules. Building a site with proton is much like panini except you use Twig instead of handlebars. The syntax of Twig is very similar to handlebars. There is some documentation at [https://docs.page/foundation/proton](url) . Proton is far from a finished product, but I have been able to convert some of my websites from panini to proton. Proton is much faster than panini. I think eventually it will be a good replacement for panini.

**Reply by clizlee** [reply]
ah ok, how do you run yarn? sorry, I know stupid question. is it just `yarn start`? I do have the directory structure setup so guess I'm almost there.

**Reply by benthien** [reply]
After init you run yarn install. The command yarn by itself defaults to yarn install.
